### PR TITLE
[azurecompute-arm] performance improvements misc

### PR DESCRIPTION
**Pull request by andreaturli**
- improve cleanup performances: by using resourceGroupApi.resources
- improve list images performance: by avoiding to get image details

/cc @nacx

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Identity.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import org.jclouds.json.SerializedNames;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class Identity {
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Identity.java]: Is there any reason why this class is not an inner class of Resource?

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/ResourceGroupApi.java [hunk 2]
```diff
@@ -76,12 +78,20 @@
    @Nullable
    ResourceGroup get(@PathParam("name") String name);
 
+   @Named("resourcegroup:resources")
+   @GET
+   @Path("/{name}/resources")
+   @SelectJson("value")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   @Nullable
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/ResourceGroupApi.java]: Given the configured fallback the returned link is never going to be null.

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Identity.java (hunk 1 above)
Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Identity.java]: not really, I will make it an inner class

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/ResourceGroupApi.java (hunk 2 above)
Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/ResourceGroupApi.java]: right!

**Comment by nacx** [comment]
@andreaturli Having a deeper look I've noticed we actually need the version properties if we want to be able to support images from the marketplace. We need to know the marketplace information from the image (done [here]( https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMImageToImage.java#L137-L145)).

Without that, deploying a node from an image from the marketplace will fail. VMs from the market place need to configure the *plan*, as done [here](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java#L162-L165).

Given that we already have an image cache and impact when listing images should be constrained to the first time they are looked up, I'd say it is more important to keep the ability to deploy marketplace images than improving the image list response time.

**Comment by andreaturli** [comment]
I think it makes sense, I will revert that second commit

**Comment by andreaturli** [comment]
rebuild please

**Comment by andreaturli** [comment]
@nacx is it ok?

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java [hunk 3]
```diff
@@ -200,8 +200,8 @@
             for (Version version : versionList) {
                Version versionDetails = osImageApi.getVersion(publisherName, offer.name(), sku.name(), version.name());
                VMImage vmImage = VMImage.azureImage().publisher(publisherName).offer(offer.name()).sku(sku.name())
-                     .version(versionDetails.name()).location(location).versionProperties(versionDetails.properties())
-                     .build();
+                       .version(versionDetails.name()).location(location).versionProperties(versionDetails.properties())
+                       .build();
                osImagesRef.add(vmImage);
             }
          }
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java]: Discard the changes in this class.

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMImageToImage.java [hunk 4]
```diff
@@ -176,10 +176,21 @@ public static Plan getMarketplacePlanFromImageMetadata(Image image) {
...
+               
+//               Version versionDetails = osImageApi.getVersion(publisherName, offer.name(), sku.name(), version.name());
+//                              VMImage vmImage = VMImage.azureImage().publisher(publisherName).offer(offer.name()).sku(sku.name())
+//                                    .version(versionDetails.name()).location(location).versionProperties(versionDetails.properties())
+//                                    .build();
+
+//               family = OsFamily.fromValue(image.versionProperties().osDiskImage().operatingSystem().toUpperCase());
             }
 
             // only 64bit OS images are supported by Azure ARM
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMImageToImage.java]: Discard the changes in this class.

Comment (andreaturli) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VMImageToImage.java]: ops

**Comment by nacx** [comment]
Pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/6b35f6fd) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/a5a4f30b). Thanks!

**Pull request closed without merging**